### PR TITLE
Adding query player language

### LINKED ISSUE #787: [Scarpet Suggestion] Querying player language
It would be nice, and it is sent in `ClientSettingsC2SPacket`. It is not really saved anywhere, but the packet is passed to the `ServerPlayerEntity` in `setClientSettings`.

I'm a total noob in Mixin, so I'll probably not try to implement this at least Soon™.

### PR DESCRIPTION
**Pull request by Ghoulboy78**
Fixes #787 
- [x] Todo test

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
Ok, this works well. It's a bit of a mixing horror show, because not only do I have to mix into the `setClientSettings` method in `ServerPlayerEntity` as mentioned [here](https://github.com/gnembon/fabric-carpet/issues/787#issue-848408862) by altrisi, I *also* have to mix into `ClientSettingsC2SPacket` as the language is private. Not sure if that means it should stay that way or what tho tbh, cos on the one hand gnembon mentioned here : 
> some players may want to remain anonymous. That's why there is 'ping' for players, but not 'ip'. Language stays in the gray zone between these two.
I understand this and respect that viewpoint, but then gnembon said this:
> I am on the fence of the language thingy, but on the 'its fine' side.
So I think we should have a poll. 👍 if you think we should keep this and allow player's language to be seen by server, 👎 if you think not.

**Comment by DragonEggBedrockBreaking** [comment]
I agreed, because language reveals little about someone, e.g. if it is a common language (e..g spanish), even the country is ambiguous (spain, argentina, chile, etc.), and if it is specific to s relaitevely small country (e.g. bulgaria), it could still be someone just practising the language or smth. You have no way to know their actual location through their language, only what language they know. The only potential issue I can think of is someone being racist to people who speak a certain language, but that is a separate issue, and there are other equally bad things that can be done without knowing the language, so this point is anyway quite invalid.

**Comment by gnembon** [comment]
lol even vanilla doesn't really use it for anything

**Comment by Ghoulboy78** [comment]
I have made the poll in carpet mod channel in Sci discord, so anyone there can vote. After a week, I will either close or leave this pr open, and let gnembon know that this pr is ok to accept or something, idrk. People may want to maintain privacy, in which case they should be allowed to decide. https://discord.com/channels/211786369951989762/573613501164159016/834157181229203516

**Comment by gnembon** [comment]
community hath spoken, go for it.

File: src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java [hunk 1]
```diff
@@ -44,6 +46,8 @@ public ServerPlayerEntity_scarpetEventMixin(World world, BlockPos blockPos, floa
 
     @Shadow public boolean notInAnyWorld;
 
+    @Shadow public int pingMilliseconds;
```
Comment (gnembon) [inline src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java:49]: what do you need it for?

File: src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java [hunk 2]
```diff
@@ -126,4 +130,16 @@ public boolean isInvalidEntityObject()
     {
         return isInvalidReference;
     }
+
+    //getting player language
+    private String language = null;
```
Comment (gnembon) [inline src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java:135]: you don't need to == null here (its gonna be added to all constructors if you do it this way)

File: src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java (hunk 1 above)
Comment (Ghoulboy78) [reply src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java:49]: Umm, Im not sure how it got there...

File: src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java (hunk 2 above)
Comment (Ghoulboy78) [reply src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java:135]: ye I noticed, just wasn't 100% sure when that method got called cos as I mentioned I just coded this up in school without being able to test, so I didn't want NPE

**Comment by Ghoulboy78** [comment]
ok then this is complete now @gnembon

**Review by gnembon — APPROVED** [review]

**Pull request merged**